app/models/nonprofit.rb

```diff
@@ -226,7 +226,7 @@ def as_json(options = {})
   end
 
   def url
-    "/#{self.state_code_slug}/#{self.city_slug}/#{self.slug}"
+    Rails.application.routes.url_helpers.slugged_nonprofit_path(self)
```

Comment:
Ultimately, Nonprofit shouldn't have this method here as this is a view concern. But it's there already and I don't know for sure all of the places its used. This just replaces it using the route helper